akeneo-design-system/src/components/Input/TagInput/TagInput.tsx

```diff
@@ -0,0 +1,223 @@
+import React, {useState, useRef, ChangeEvent, FC, RefObject, KeyboardEvent, useCallback} from 'react';
+import styled from 'styled-components';
+import {AkeneoThemedProps, getColor} from '../../../theme';
+import {CloseIcon} from '../../../icons';
+import {Key} from '../../../shared';
+
+type TagInputProps = {
+  /**
+   * Specifies if the component will accept duplicated tags or not
+   */
+  allowDuplicates: boolean;
+
+  /**
+   * Tags to display
+   */
+  tags: string[];
+
+  /**
+   * Handle called when tags are updated
+   */
+  setTags: (tags: string[]) => void;
+
+  /**
+   * Placeholder displayed where there is no tag
+   */
+  placeholder?: string;
+
+  /**
+   * Defines if the input is valid on not.
+   */
+  isInvalid?: boolean;
+};
+
+const MAX_TAGS = 100;
+
+const TagInput: FC<TagInputProps> = ({allowDuplicates, setTags, placeholder, isInvalid, tags = []}) => {
+  const [isLastTagSelected, selectLastTag] = useState<boolean>(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const containerRef = useRef<HTMLUListElement>(null);
+  const inputContainerRef = useRef<HTMLLIElement>(null);
+
+  const updateTags = useCallback(
+    (updatedTags: string[]) => {
+      updatedTags = updatedTags.slice(0, Math.min(MAX_TAGS, updatedTags.length));
+      setTags(updatedTags);
+    },
+    [setTags]
+  );
+
+  const onChangeCreateTags = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const tagsAsString = event.currentTarget.value;
+      if (tagsAsString !== '') {
+        let newTags = tagsAsString.split(/[\s,;]+/);
+        if (newTags.length === 1) {
+          return;
+        }
+        newTags = newTags.filter((tag: string) => tag.trim() !== '');
+
+        createTags([...tags, ...newTags]);
+      }
+    },
+    [tags]
+  );
+
+  const onBlurCreateTag = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const inputCurrentValue = event.currentTarget.value.trim();
+      if (inputCurrentValue !== '') {
+        createTags([...tags, ...[inputCurrentValue]]);
+      }
+    },
+    [tags]
+  );
+
+  const createTags = useCallback(
+    (newTags: string[]) => {
+      if (!allowDuplicates) {
+        newTags = arrayUnique(newTags);
+      }
+      updateTags(newTags);
+      if (inputRef && inputRef.current) {
+        inputRef.current.value = '';
+      }
+    },
+    [arrayUnique, inputRef, updateTags]
+  );
+
+  const removeTag = useCallback(
+    (tagIdToRemove: number) => {
+      const clonedTags = [...tags];
+      clonedTags.splice(tagIdToRemove, 1);
+      updateTags(clonedTags);
+    },
+    [tags, updateTags]
```

Comment:
```suggestion
    (tagIndexToRemove: number) => {
      const clonedTags = [...tags];
      clonedTags.splice(tagIndexToRemove, 1);
      updateTags(clonedTags);
    },
    [tags, updateTags]
```

Id feels weird as this is not an identifier but an index in the array